app/models/relationship.rb

```diff
@@ -159,6 +159,13 @@ def create_or_update_parent_id
   def self.create_unless_exists(source_id, target_id, relationship_type, options = {})
     # Verify that the target is not part of another source; in this case, we should return the existing relationship.
     r = Relationship.where(target_id: target_id).where.not(source_id: source_id).last
+    # should update options if exists
+    unless r.nil? || !options.blank?
+      options.each do |key, value|
+        r.send("#{key}=", value) if r.respond_to?("#{key}=")
+      end
+      r.save!
+    end
```

Comment:
Sawy, shouldn't we update the options only if the current values for the `Relationship` object are blank?

Reply:
Hum, I see, good point. But we should at least make sure that we're not _erasing_ previous values. So I suggest you check if at least one of the `value`'s from `options` is not blank.

Reply:
Done and check against model field